mcs/class/System/System.Net.NetworkInformation/Ping.cs

```diff
@@ -215,7 +215,7 @@ static IPAddress GetNonLoopbackIP ()
 					return addr;
 #pragma warning restore 618
 
-			throw new InvalidOperationException ("Could not resolve non-loopback IP address for localhost");
+			return IPAddress.Loopback;
```

Comment:
I don't know why we need a non-loopback address, but this is not the correct way of fixing it.
You need to modify SendPrivileged() so it doesn't rely on GetNonLoopbackIP().